Readme plugin (#56)
weakish/weakish.github.com: plugins/readme/README.md

> # lume-plugin-readme

Author: ```suggestion
# readme
```

> A Lume plugin that transforms URLs of homepage files (like `README.md`) to clean directory URLs, enabling GitHub Pages-style documentation sites.

Author: ```suggestion
A Lume plugin that uses `README.md` or specified page as homepage of the website, as on GitHub Pages and docsify.
```

## Problem

By default, Lume generates URLs like this:

```text
/docs/getting-started/README.md  →  /docs/getting-started/README/
```

This plugin transforms them to:

```text
/docs/getting-started/README.md  →  /docs/getting-started/
```

## Usage

### Basic usage

Transform all `README` files to directory URLs:

```ts
import readme from "./plugins/readme/mod.ts";

site.use(readme());
```

### Custom homepage basenames

Specify a custom ordered array of basenames to transform:

```ts
site.use(readme({
  homepage: ["home", "index", "README"],
}));
```

Do not include file extensions — Lume's `srcPath` is extension-less.
The first match in the array wins. Matching is case insensitive.

### Exclude paths

Skip transformation for specific directories:

```ts
site.use(readme({
  exclude: ["/blog/", "/notes/"],
}));
```

### Include only specific paths

Only transform files in specified directories:

```ts
site.use(readme({
  include: ["/docs/", "/guides/"],
}));
```

> **Note:** `exclude` and `include` cannot be used together.

## Options

| Option | Type | Default | Description |
|--------|------|---------|-------------|

> | `homepage` | `string[]` | `["README"]` | Ordered array of homepage basenames (case insensitive, no extensions) |

Author: ```suggestion
| `homepage` | `string[]` | `["README"]` | Ordered array of homepage basenames (case insensitive, no extensions, first match wins) |
```
